src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -363,21 +362,25 @@ type AccessStrategy<'event, 'state> =
     /// doing a <c>fold</c> from any such event will yield a correct and complete state
     /// In other words, the <c>fold</c> function should not need to consider either the preceding <c>'state</c> or <c>'event</c>s.
     | LatestKnownEvent
-    /// Ensures a snapshot/compaction event from which the state can be reconstituted upon decoding is always present
-    /// (embedded in a special snapshot stream as an event), generated every <c>batchSize</c> events using the supplied
-    /// <c>toSnapshot</c> function.
-    /// When loading the stream it'll first fetch the snapshot.
-    | RollingSnapshots of originType: string * toSnapshot : ('state -> 'event)
+    /// <summary>
+    /// Generates and stores a snapshot event in an adjacent <c>{category}:snapshot-{stream_id}</c> stream
+    /// The generation happens every <c>batchSize</c> events.
+    /// This means the state of the stream can be reconstructed with exactly
+    /// 2 round-trips to the database.
+    /// The first round-trip fetches the most recent event of type <c>snapshotType</c> from the snapshot stream.
+    /// The second round-trip fetches <c>batchSize</c> events from the position of the snapshot
+    /// The snapshot is generated with the supplied <c>toSnapshot</c> function
+    /// </summary>
+    | AdjacentSnapshots of snapshotType: string * toSnapshot : ('state -> 'event)
```

Comment:
snapshotType still needs a better name that conveys its the caseName associated with a snapshot DU case (ideally the same one as `toSnapshot`).

🤔 I wonder if could it be inferred from an Expr<'event> that creates a dummy value e.g. `<@ fun () -> Snapshot Unchehecked.defaultof<_> @>`

Reply:
> 🤔 I wonder if could it be inferred from an Expr<'event> that creates a dummy value e.g. <@ fun () -> Snapshot Unchehecked.defaultof<_> @>

That might be a bit much to ask of a consumer of the library

> snapshotType still needs a better name that conveys its the caseName associated with a snapshot DU case (ideally the same one as toSnapshot).

I'll ask around what people think makes sense, for now I've renamed to "snapshotEventCaseName"

Reply:
>That might be a bit much to ask of a consumer of the library

yes, was trying to provoke thought by putting something pejorative out, but there has to be some static reflection type trick of some kind!
I personally would use a `[<>]` literal in the `module Events` and use to it in Fold.snapshoteventcasename if it arose, but then I'm not the person that will whack it in and not understand its role, either